databricks_connector.py

```diff
@@ -0,0 +1,663 @@
+            token=self._token,
+        )
+
+        # Override user agent to {ISV Name}-{Product Name} per Databricks best practices
+        api_client.default_headers['user-agent'] = 'splunk-soar'
+
+        return api_client
+
+    def _handle_test_connectivity(self, param):
+        self.save_progress(consts.TEST_CONNECTIVITY_PROGRESS_MESSAGE)
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        try:
+            api_client = self._get_api_client()
+            dbfs_service = DbfsService(api_client)
+            dbfs_service.get_status(consts.TEST_CONNECTIVITY_FILE_PATH)
+
+            self.save_progress(consts.TEST_CONNECTIVITY_SUCCESS_MESSAGE)
+            return action_result.set_status(phantom.APP_SUCCESS)
+        except Exception as e:
+            error_message = self._get_error_msg_from_exception(e)
+            self.save_progress(error_message)
+            return action_result.set_status(phantom.APP_ERROR, consts.TEST_CONNECTIVITY_ERROR_MESSAGE)
+
+    def _handle_create_alert(self, param):
+        self.debug_print(f'In action handler for: {self.get_action_identifier()}')
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        data = {}
+        data['name'] = param['name']
+        data['query_id'] = param['query_id']
+
+        options = {
+            'column': param['column'],
+            'op': param['operator'],
+            'value': param['value'],
+        }
+        self._set_key_if_param_defined(options, param, 'custom_body')
+        self._set_key_if_param_defined(options, param, 'custom_subject')
+        self._set_key_if_param_defined(options, param, 'muted')
+        self._set_key_if_param_defined(options, param, 'schedule_failures')
+        data['options'] = options
+
+        self._set_key_if_param_defined(data, param, 'rearm')
+        self._set_key_if_param_defined(data, param, 'parent')
+
+        try:
+            api_client = self._get_api_client()
+            result = api_client.perform_query(**DatabricksEndpoint.CREATE_ALERT.api_info, data=data)
+            action_result.add_data(result)
+            summary = {
+                'status': consts.CREATE_ALERT_SUCCESS_MESSAGE,
+            }
+            action_result.update_summary(summary)
+            return action_result.set_status(phantom.APP_SUCCESS)
+
+        except Exception as e:
+            error_message = self._get_error_msg_from_exception(e)
+            self.save_progress(error_message)
+            return action_result.set_status(phantom.APP_ERROR, consts.CREATE_ALERT_ERROR_MESSAGE, error_message)
+
+    def _handle_list_alerts(self, param):
+        self.debug_print(f'In action handler for: {self.get_action_identifier()}')
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        try:
+            api_client = self._get_api_client()
+            api_info = DatabricksEndpoint.LIST_ALERTS.api_info_with_interpolation()
```

Comment:
you can also just do
```suggestion
            api_info = DatabricksEndpoint.LIST_ALERTS.api_info
```
for cases like this which don't have any variables to interpolate. I think it's understandable either way, so up to you, but just pointing it out :)

Reply:
Oh, I didn't realize that - I'll change it for those actions.